### ISSUE TITLE
Week 11 Discussion

### ISSUE THREAD

**Issue by kayvee** [stefanbund/cis155#17]
> For this week's discussion topic, please read chapter 5 in your text, if you have not already.  After reading chapter 5 please answer this topic.
> 
> Chapter 5 provides several types of conditional statements - if, if-else, while and others. Certain situations demand for each type of conditional statement. Can you provide an example of when each type of conditional statement is recommended? 
> 
> Treat your answer as a form of playbook, indicating when certain types of programming statements are most appropriate.  Also, feel free to post multiple times, explaining a new situation each post. Also, be sure to reply to the posts of others. 

**while** checks condition at the beginning of the loop to see if it will execute the loop

This establishes a loop with a conditional that will only be executed if a condition needs to be met first.

For example, if you write a game where someone must guess a number. You should first check that number and see the number doesn't match. If it doesn't, you should enter the loop and continue asking to see if the person can guess the number. Once they do, the loop is exited and you can move onto a statement where the program tells them they've won.

**do-while** checks condition at the end of the loop to see if it will continue executing loop (i.e., it always executes at least once)

This loop can work for the same example above, but instead of asking for the number before the while loop and checking, you ask for it in the 'do' and check it at the end to see if the number _is_ correct. Once it is, the loop isn't entered anymore.

These were the only conditionals in chapter 5.

**Comment by Irfan009** [comment]
_For this week's discussion topic, please read chapter 5 in your text, if you have not already. After reading chapter 5 please answer this topic.

Chapter 5 provides several types of conditional statements - if, if-else, while and others. Certain situations demand for each type of conditional statement. Can you provide an example of when each type of conditional statement is recommended?

Treat your answer as a form of playbook, indicating when certain types of programming statements are most appropriate. Also, feel free to post multiple times, explaining a new situation each post. Also, be sure to reply to the posts of others._

**While loops** allow executing a certain block of code multiple time by writing it only once. The while loop extends the idea of an If statement by executing the body again and again and looking at the test each time vs. looking at the test once and if the condition is true then may be executing the body once.
**Syntax:**
While (expression)
 {
        Statement(s)
}
Example:

public class Mainroom {
public static void main(String[] args) 
        {
          int counter =0;
          while(counter <=5)
        {
          System.out.println(counter);
            counter ++;
        }
        System.out.println("All Done!");
        }
}

![screen shot 2016-04-17 at 1 14 26 pm](https://cloud.githubusercontent.com/assets/16991525/14589908/6bff734c-04a2-11e6-9392-5982ad179fd9.png)

**DO while loop** is pretty much the same as a while loop , except the condition in a Do While loop is always tested at the bottom of the code instead of the top. As a result the statements with in the Do block will execute at least once.
*_Syntax: *_       Do   {
            Statement(s)
}
While (expression)

*_Example:   *_  
public class Mainroom {
public static void main(String[] args) 
        {
         int counter =0;
         do
        {
        System.out.println("The Counter is at :"+counter)
counter ++;
        }
        while(counter <=5);
        System.out.println("All Done!");
}
}
![screen shot 2016-04-17 at 1 32 25 pm](https://cloud.githubusercontent.com/assets/16991525/14589910/7451e20a-04a2-11e6-885d-af7b349b65f2.png)

~Irfan Awan

**Comment by stefanbund** [comment]
When it comes to the use of while, wouldn't you expect to find the while at the beginning of the application? Isn't while best used to monitor the value of a variable, then instruct the program based on its value?

From Stefan

**Comment by stefanbund** [comment]
Wouldnt you expect the while statement to be used rather infrequently, whereas if and if else are expected to be used constantly?

From Stefan

**Comment by kayvee** [comment]
Absolutely. I think that was what I was trying to establish, but that the
'while' can be used in both cases in different ways.

**Comment by forrestbennett** [comment]
```
For this week's discussion topic, please read chapter 5 in your text, if you have not already. After reading chapter 5 please answer this topic.

Chapter 5 provides several types of conditional statements - if, if-else, while and others. Certain situations demand for each type of conditional statement. Can you provide an example of when each type of conditional statement is recommended?

Treat your answer as a form of playbook, indicating when certain types of programming statements are most appropriate. Also, feel free to post multiple times, explaining a new situation each post. Also, be sure to reply to the posts of others.
```

**While loop** is executed repeatedly based on a given condition. It will loop while the condition is true.
**Do-while** is the same thing only it is called upon at the bottom of the loop so it is always executed at least once. 
**## while**
 int count = 1;
        while (count < 11) {
            System.out.println("Count is: " + count);
            count++;

**## Do-while**
int count = 1;
        do {
            System.out.println("Count is: " + count);
            count++;
        } while (count < 11);

**Comment by stefanbund** [comment]
Good job in making this contrast. I am interested in what kinds of situations you might use it in. Such as if you had a variable which needed to be observed during the run time of an application. This variable might relate to whether the user is authenticated, or whether they have changed a meaningful preference. 

Couldn't the application behave differently based upon these environmental values?

From Stefan

**Comment by kayvee** [comment]
@Irfan009 :

> The while loop extends the idea of an If statement by executing the body again and again and looking at the test each time vs. looking at the test once and if the condition is true then may be executing the body once.

I think you make an interesting point by trying to tie 'if' into this, but I think it goes a little bit further than that. 'if' is control flow based on an element, making a decision on that value, sure. But it does not usually go for 'running,' if that makes sense, the same way 'while' does. 'While' is more useful in updating and building in a program, whereas if will be used to set 'one' aspect instead.

**Comment by stefanbund** [comment]
Nice contrast here. I like to associate the while loop with more global  environmental   decisions. If can be executed thousands of times within a while. I like that one to 1000 ratio in regards to while versus if 

this perhaps provides one element to our playbook for flow control.

From Stefan
